bundles/org.eclipse.ui.ide/extensions/org/eclipse/ui/part/ResourceTransfer.java

```diff
@@ -129,25 +129,21 @@ protected void javaToNative(Object data, TransferData transferData) {
 
 		int resourceCount = resources.length;
 
-		try {
-			ByteArrayOutputStream out = new ByteArrayOutputStream();
-			DataOutputStream dataOut = new DataOutputStream(out);
+		ByteArrayOutputStream out = new ByteArrayOutputStream();
```

Comment:
Shouldn't out variable be handled by try-with-resources too? With that change it's no longer closed but it used to be in the old code.

Reply:
I guess this is because `ByteArrayOutputStream` does not need to be closed (`close()` actually is a no-op there).

Reply:
see ByteArrayOutputStream#close - its only memory not an actual resource